zerver/lib/actions.py

```diff
@@ -2883,7 +2883,9 @@ def notify_realm_emoji(realm):
     send_event(event, user_ids)
 
 def do_add_realm_emoji(realm, name, img_url):
-    RealmEmoji(realm=realm, name=name, img_url=img_url).save()
+    emoji = RealmEmoji(realm=realm, name=name, img_url=img_url)
+    emoji.full_clean()
```

Comment:
Hmm, we haven't used `full_clean` before; I'll need to look into that feature...